src/hotspot/share/gc/g1/g1BarrierSet.hpp

```diff
@@ -53,6 +53,10 @@ class G1BarrierSet: public CardTableBarrierSet {
   G1BarrierSet(G1CardTable* table);
   ~G1BarrierSet() { }
 
+  bool card_mark_must_follow_store() const {
```

Comment:
Should be "virtual".  (Probably should be "override", but nobody's gotten around to vetting and proposing that change to the style guide.)